### PR TITLE
Add DataContextService - Database integration

### LINKED ISSUE #31: [DEV] Database integration
**Summary:** Integrate the project with a MongoDB engine.

In order to be able to create bots, commands, and other entities, it is necessary to integrate the software with a database.

**Functional notes:**
- The integration is only necessary for the backend project. The resources one contains only resources, and the worker project does not need to access data.

**Technical notes:**
- A service must be created so it is easy to inject the database connection anywhere in the project.
- DTO classes will be necessary to extend the functionality of the resources ones.
- The service must be generic, so all kinds of data can be managed.
- Keep the documentation in mind.
  - [DB](https://github.com/harvestcore/tfm-doc/blob/develop/md/dise%C3%B1o/base-datos.md)
  - [Tools](https://github.com/harvestcore/tfm-doc/blob/develop/md/analisis/herramientas.md)

**Relates to:**
- #1
- #5

**Dependencies:**
- #30

**Subtasks:**
- None for now.

**Acceptance criteria:**
- Unit tests are added if possible.
- The technical notes are met.

### PR DESCRIPTION
**Pull request by harvestcore**

### CODE REVIEW

**Comment by JJ** [comment]
Every PR needs to have a description. At least the issues it addresses, and the technical decisions that have been taken. That said, it's probably too early for database integration.

File: .github/workflows/test-dotnet.yml [hunk 1]
```diff
@@ -32,6 +35,11 @@ jobs:
         run: ls -lah
         working-directory: ${{ env.WORKING_DIR }}
 
+      - name: Start MongoDB
```
Comment (JJ) [inline .github/workflows/test-dotnet.yml:38]: This is *really* premature. You need to document all technical decisions you're taking. If you've chosen MongoDB, you need to say why and what is the state of the art in this area. But, as I say in #35, you need to organize development in progressively complex products; you can't even think about storage until you've designed and tested all relevant business logic.
Also, anything related to data storage needs to be mocked for testing. No need to start a real database for unit tests.

File: backend/Matroos.Backend.sln [hunk 2]
```diff
@@ -13,6 +13,10 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{AB62B8E2-AB3
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "tests", "tests", "{63D2FA29-193B-4880-B4F3-4E8AF0DF78E1}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Matroos.Backend.Core", "src\Matroos.Backend.Core\Matroos.Backend.Core.csproj", "{4B396C9B-A5D0-4BA5-9307-84DD207F4C46}"
```
Comment (JJ) [inline backend/Matroos.Backend.sln:16]: This file needs to be in the repo?

File: backend/src/Matroos.Backend.Core/Classes/QueryFilter.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+﻿namespace Matroos.Backend.Core.Classes;
+
+public class QueryFilter
+{
+    /// <summary>
+    /// Documents to be skip.
+    /// </summary>
+    public int Skip { get; set; } = 0;
```
Comment (JJ) [inline backend/src/Matroos.Backend.Core/Classes/QueryFilter.cs:8]: You *never, ever* use getters and setters of everything unless a user story requires you to do so. Will this change after construction? Is this an object value or an entity? How has this been designed according to domain driven design (or other methodology)?

File: backend/src/Matroos.Backend.Core/Services/DataContextService.cs [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+using Matroos.Backend.Core.Classes;
+using Matroos.Backend.Core.Dto;
+using Matroos.Backend.Core.Services.Interfaces;
+using Matroos.Resources.Services.Interfaces;
+
+using MongoDB.Driver;
+
+namespace Matroos.Backend.Core.Services;
+
+public class DataContextService : IDataContextService
```
Comment (JJ) [inline backend/src/Matroos.Backend.Core/Services/DataContextService.cs:12]: I guess you have this for possible injection, nice.

**Review by JJ — CHANGES_REQUESTED** [review]
I have checked a file, and it consists of a few commits, none of which reference any issue, and thus any user story. *All* changes to the code need to be made guided by them; it's the only way to guarantee they add value to the client.
Biggest problem, however, is that I think adding data services is very premature. This PR should be done towards the end of the development. All business logic needs to be tested first.

**Comment by JJ** [comment]
Is this left for later?

**Comment by harvestcore** [comment]
> Is this left for later?

@JJ, yes. I've changed the PR to draft.

**Comment by harvestcore** [comment]
Outdated, #31 is resolved in #89

**Pull request closed without merging**